### PR TITLE
Prevent need to restart Faucet

### PR DESCRIPTION
**Pull request by hewison-chris**
Recently in PROD env it has been required that the Faucet is restarted
as it can get into a state where it filters out all sent utxos and none
remain. The restart clears the sent_utxo and it then resumes ok. This
fix will clear the sent_utxo when no more utxo are available. It might
cause some double spend but at least it will continue to send txs.

### CODE REVIEW

File: source/faucet/main.d [hunk 1]
```diff
@@ -252,7 +252,8 @@ public class Faucet : FaucetAPI
                 this.state.sent_utxos.put(kv.key);
                 return TxBuilder(kv.value.output, kv.key);
             })
-            .map!(txb => txb.unlockSigner(&this.keyUnlocker).split(
+            .map!(txb => txb.unlockSigner(&this.keyUnlocker)
+                .split(
```
Comment (linked0) [inline source/faucet/main.d:256]: I don't think this is not related to this commit.

**Review by linked0 — APPROVED** [review]
LGTM

File: source/faucet/main.d (hunk 1 above)
Comment (hewison-chris) [reply source/faucet/main.d:256]: I think you mean
`I think this is not related to this commit.`

Comment (hewison-chris) [reply source/faucet/main.d:256]: I will split into separate commit.
